src/app/payments/get-protocol-fee.ts

```diff
@@ -167,7 +166,7 @@ const estimateLightningFee = async ({
       : "undefined",
   })
 
-  const priceRatio = WalletPriceRatio({ usd: usdPaymentAmount, btc: btcPaymentAmount })
+  const priceRatio = await builder.walletPriceRatio()
```

Comment:
is this builder responsability? we use builder calling specific methods to get a final/structured object at the end.. I understand this but not sure if should be part of builder

Reply:
It's reading values out at an intermediate state of the builder which we already do for payment amounts as well.